torch/testing/_internal/composite_compliance.py

```diff
@@ -0,0 +1,172 @@
+        'vsplit',
+        'hsplit',
+        'tensor_split',
+        'chunk',
+        'swapaxes',
+        'slice',
+        '_reshape_alias',
+        '_unsafe_view',
+        '_conj',
+        'alias',
+    }
+
+def is_inplace_view_fn(func):
+    return func.__name__ in {
+        'squeeze_',
+        'unsqueeze_',
+        'transpose_',
+        't_',
+    }
+
+class CompositeCompliantTensor(torch.Tensor):
+    elem: torch.Tensor
+
+    __slots__ = ['elem']
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        # The storage of CompositeCompliantTensor should never be used directly
+        # by a CompositeImplicitAutograd operation; if the CompositeImplicitAutograd
+        # operator attempts to read from the storage without dispatching then it'll
+        # raise a RuntimeError due to it being a meta storage.
+        r = torch.Tensor._make_wrapper_subclass(  # type: ignore[attr-defined]
+            cls, elem.size(),
+            dtype=elem.dtype, layout=elem.layout,
+            device=elem.device, requires_grad=elem.requires_grad)
+        r.elem = elem
+        return r
+
+    def __repr__(self):
+        return f"CompositeCompliantTensor({self.elem})"
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        def unwrap(e):
+            return e.elem if isinstance(e, CompositeCompliantTensor) else e
+
+        def wrap(e):
+            return CompositeCompliantTensor(e) if isinstance(e, torch.Tensor) else e
+
+        if func.__name__ in ('set_', 'resize_'):
+            raise RuntimeError(
+                f"{func.__name__} is not allowed to be called inside of "
+                f"CompositeImplicitAutograd operators.")
+
+        with no_dispatch():
+            unwrapped_args = tree_map(unwrap, args)
+            unwrapped_kwargs = tree_map(unwrap, kwargs)
+            unwrapped_rs = func(*unwrapped_args, **unwrapped_kwargs)
+            rs = tree_map(wrap, unwrapped_rs)
+
+        if is_view_fn(func):
+            # Autograd asserts that for B = A.view_fn(...), B and A's storages
+            # are the same. Here we try to make B alias A to avoid those asserts.
+            with no_dispatch():
+                # Idea: this is a weird way of getting a storage that aliases the input.
+                # 1. under no_dispatch, all of the wrapper tensors look like meta tensors
+                # 2. we run func, which ends up running the meta tensor version of the view op
+                # 3. the result is one or more meta tensor that aliases the input.
+                # 4. we set the storage of the wrapper tensor results to be
+                #    the meta tensor(s) that aliases the input
+                result = func(*args, **kwargs)
+                if isinstance(result, tuple) or isinstance(result, list):
+                    for a, b in zip(rs, result):
+                        a.set_(b)
+                else:
+                    rs.set_(result)
```

Comment:
@albanD, @bdhirsh: if we think the "make the wrapper output's storage alias the wrapper input's storage" logic is too hacky and if we're willing to accept that these tests do not pass in DEBUG mode while we figure out a better solution (most of `test_python_dispatch` fails under DEBUG mode) then I can delete this code for now

Reply:
To me, this seems like the starting point for a proper fix to #65339. So... it's OK for now, let's just keep making it better?